x-pack/plugins/endpoint/public/applications/endpoint/store/policy_list/middleware.ts

```diff
@@ -26,20 +29,29 @@ export const policyListMiddlewareFactory: MiddlewareFactory<PolicyListState> = c
         pageIndex = state.pageIndex;
       }
 
-      // Need load data from API and remove fake data below
-      // Refactor tracked via: https://github.com/elastic/endpoint-app-team/issues/150
-      const { getFakeDatasourceApiResponse } = await import('./fake_data');
-      const { items: policyItems, total } = await getFakeDatasourceApiResponse(pageIndex, pageSize);
+      try {
+        const { items: policyItems, total } = await sendGetEndpoingDatasources(http, {
+          query: {
+            perPage: pageSize,
+            page: pageIndex + 1,
+          },
+        });
 
-      dispatch({
-        type: 'serverReturnedPolicyListData',
-        payload: {
-          policyItems,
-          pageIndex,
-          pageSize,
-          total,
-        },
-      });
+        dispatch({
+          type: 'serverReturnedPolicyListData',
+          payload: {
+            policyItems,
+            pageIndex,
+            pageSize,
+            total,
+          },
+        });
+      } catch (err) {
```

Comment:
Since `err` will be any, the dispatch call accepts it even though the runtime value may not satisfy the expected payload type. 

![image](https://user-images.githubusercontent.com/35559/77080123-5006ef80-69cf-11ea-8447-63f3b7751ff7.png)

Reply:
I can't just pass the `err` through because the `ApiServerError` structure is in the `body` property of the error object returned by `http.*` methods, thus why I do the check below.  I test this by adding a `kuery` param on the fetch above to make it fail 😞